test/README.md

```diff
@@ -166,6 +166,47 @@ namespace := test.AppendRandomString('arendelle')
 
 _See [randstring.go](./randstring.go)._
 
+#### Check Pipeline resources
+
+After creating Pipeline resources or making changes to them, you will need to wait for the system
+to realize those changes. You can use polling methods to check the resources reach the desired state.
+
+The `WaitFor*` functions use the kubernetes [`wait` package](https://godoc.org/k8s.io/apimachinery/pkg/util/wait).
+To poll they use [`PollImmediate`](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#PollImmediate)
+and the return values of the function you provide behave the same as
+[`ConditionFunc`](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ConditionFunc):
+a `bool` to indicate if the function should stop or continue polling, and an `error` to indicate if
+there has been an error.
+
+For example, you can poll a `TaskRun` object to wait for it to have a `Status.Condition`:
+
+```go
+// Verify status of TaskRun (wait for it)
+err = WaitForTaskRunState(c, hwTaskRunName, func(tr *v1alpha1.TaskRun) (bool, error) {
+    if tr.Status != nil && len(tr.Status.Conditions) > 0 {
+        return true, nil
+    }
+    return false, nil
+}, "TaskRunHasCondition")
+```
+
+_[Metrics will be emitted](#emit-metrics) for these `Wait` method tracking how long test poll for._
+
+We also have `Check*` variants of many of these methods with identical signatures, same example:
+
+```go
+var revisionName string
+err := test.CheckConfigurationState(clients.ServingClient, configName, func(c *v1alpha1.Configuration) (bool, error) {
```

Comment:
ConfigurationState doesn't belong in this repo

Reply:
lol whoa whoops, totally didn't copy paste that in :sweat_smile: